Mathlib/Order/ConditionallyCompleteLattice/Finset.lean

```diff
@@ -231,6 +240,21 @@ lemma sup_univ_eq_ciSup [Fintype ι] (f : ι → α) : univ.sup f = ⨆ i, f i :
     (Finset.sup_le fun _ _ => le_ciSup (finite_range _).bddAbove _)
     (ciSup_le' fun _ => Finset.le_sup (mem_univ _))
 
+theorem sup_eq_ciSup [Nonempty β] (s : Finset β) (f : β → α) :
+    s.sup f = ⨆ (b : β) (_ : b ∈ s), f b := by
```

Comment:
Can you golf this proof by first proving that both the LHS and RHS are least upper bounds for `f '' s` then using `IsLUB.unique`?

Reply:
I don't see how to do this with the existing API.